Documentation/botr/readytorun-format.md

```diff
@@ -326,6 +382,49 @@ This section marks region that contains thunks for ``READYTORUN_HELPER_DelayLoad
 helper. It is used by debugger for step-in into lazily resolved calls. It should not be required when 
 debuggers are able to handle debug info stored separately.
 
+## READYTORUN_SECTION_AVAILABLE_TYPES
+
+This section contains a native hashtable of all defined & export types within the compilation module. The key is the full type name, the value is the exported type or defined type token row ID left-shifted by one and or-ed with bit 0 defining the token type:
+
+| Bit value | Token type    |
+|----------:|:--------------|
+|         0 | defined type  |
+|         1 | exported type |
+
+The version-resilient hashing algorithm used for hashing the type names is implemented in [vm/versionresilienthashcode.cpp](https://github.com/dotnet/coreclr/blob/ec2a74e7649f1c0ecff32ce86724bf3ca80bfd46/src/vm/versionresilienthashcode.cpp#L75).
+
+## READYTORUN_SECTION_INSTANCE_METHOD_ENTRYPOINTS
+
+This section contains a native hashtable of all generic method instantiations compiled into the R2R executable. The key is the method instance signature; the appropriate version-resilient hash code calculation is implemented in [vm/versionresilienthashcode.cpp](https://github.com/dotnet/coreclr/blob/ec2a74e7649f1c0ecff32ce86724bf3ca80bfd46/src/vm/versionresilienthashcode.cpp#L128); the value, represented by the ``EntryPointWithBlobVertex`` class, stores the method index in the runtime function table, the fixups blob and a blob encoding the method signature.
+
+## READYTORUN_SECTION_INLINING_INFO
+
+**TODO**: document inlining info encoding
+
+## READYTORUN_SECTION_PROFILEDATA_INFO
+
+**TODO**: document profile data encoding
+
+## READYTORUN_SECTION_MANIFEST_METADATA
+
+Manifest metadata is an [ECMA-335] metadata blob containing extra reference assemblies introduced by inlining on top of assembly references stored in the input MSIL. This can only happen for non-versionable methods or for methods within the same *version bubble*.
```

Comment:
Do we actually use this for non-versionable methods? I do not think that this blob as designed would be sufficient for non-versionable methods outside current version bubble.

Reply:
Hmm, you're right this merits further clarification. I meant to say that inlining beyond the version bubble boundary only happens in the presence of non-versionable methods but it's true that we cannot use the manifest metadata for expressing tokens used by the non-versionable method from outside of the version bubble as we cannot rely on row ID's in external assemblies beyond the version bubble boundary to remain constant.

Reply:
[Hopefully] improved in 9th commit.